flash/video/classification/utils.py

```diff
@@ -0,0 +1,116 @@
+
+        # If a RandomSampler is used we need to pass in a custom random generator that
+        # ensures all PyTorch multiprocess workers have the same random seed.
+        self._video_random_generator = None
+        if video_sampler == torch.utils.data.RandomSampler:
+            self._video_random_generator = torch.Generator()
+            self._video_sampler = video_sampler(self._labeled_videos, generator=self._video_random_generator)
+        else:
+            self._video_sampler = video_sampler(self._labeled_videos)
+
+        self._video_sampler_iter = None  # Initialized on first call to self.__next__()
+
+        # Depending on the clip sampler type, we may want to sample multiple clips
+        # from one video. In that case, we keep the store video, label and previous sampled
+        # clip time in these variables.
+        self._loaded_video_label = None
+
+    @property
+    def video_sampler(self):
+        """
+        Returns:
+            The video sampler that defines video sample order. Note that you'll need to
+            use this property to set the epoch for a torch.utils.data.DistributedSampler.
+        """
+        return self._video_sampler
+
+    @property
+    def num_videos(self):
+        """
+        Returns:
+            Number of videos in dataset.
+        """
+        return len(self.video_sampler)
+
+    def __next__(self) -> dict:
+        """Retrieves the next clip based on the clip sampling strategy and video sampler.
+
+        Returns:
+            A dictionary with the following format.
+
+            .. code-block:: text
+
+                {
+                    'video': <video_tensor>,
+                    'label': <index_label>,
+                    'video_label': <index_label>
+                    'video_index': <video_index>,
+                }
+        """
+        if not self._video_sampler_iter:
+            # Setup MultiProcessSampler here - after PyTorch DataLoader workers are spawned.
+            self._video_sampler_iter = iter(MultiProcessSampler(self._video_sampler))
+
+        # for i_try in range(self._MAX_CONSECUTIVE_FAILURES):
+        # Reuse previously stored video if there are still clips to be sampled from
+        # the last loaded video.
+        video_index = next(self._video_sampler_iter)
+        video_tensor, info_dict = self._labeled_videos[video_index]
+        self._loaded_video_label = (video_tensor, info_dict, video_index)
+
+        sample_dict = {
+            "video": self._loaded_video_label[0],
+            "video_name": f"video{video_index}",
+            "video_index": video_index,
+            "label": info_dict,
+            "video_label": info_dict,
+        }
+        if self._transform is not None:
+            sample_dict = self._transform(sample_dict)
+
+            # # User can force dataset to continue by returning None in transform.
+            # TODO: @krshrimali
+            # if sample_dict is None:
+            #     continue
```

Comment:
If we drop transform then this can be removed

Reply:
Good point, thanks Ethan! Removed `transform` and relevant part of code.